insights/collect.py

```diff
@@ -473,6 +480,40 @@ def _parse_broker_exceptions(broker, exceptions_to_report):
     return errors
 
 
+def _write_out_blacklisted_specs(broker, output_path):
+    """
+    Write out the blacklisted specs to blacklisted_specs.txt, and create
+    a meta-data file for this file. That way it can be loaded when the
+    archive is processed.
+
+    Args:
+        broker (Broker): Broker object used for core collection.
+        output_path (str): Path of the output directory.
+    """
+    with open(os.path.join(output_path, "blacklisted_specs.txt"), "w") as of:
+        of.write(", ".join(broker.blacklisted_specs))
```

Comment:
@ryan-blakley what if you converted this to JSON instead of creating a string here that must be re-parsed?

Reply:
@bfahr I updated it to output json, and got it working on both core collection and non-core collection and confirmed that the specs and parser works from within a rule.